core/src/main/java/org/jdbi/v3/core/Jdbi.java

```diff
@@ -332,6 +330,29 @@ public HandleCallbackDecorator getHandleCallbackDecorator() {
         return this.handleCallbackDecorator.get();
     }
 
+    /**
+     * Returns the internal {@link HandleScope} object. The Jdbi instance uses this to provide handles in a given scope.
+     * The default scope is <i>per-thread</i>, so every thread manages its own handle.
+     * <br>
+     * <b>This is an internal method and not part of the public API!</b>
+     * @return A {@link HandleScope} object
+     * @param <T> type parameter
+     */
+    @Alpha
+    public final <T extends HandleScope> T getHandleScope() {
```

Comment:
Do we actually want this `<T>` return type? Why not just return the interface? For callers to actually work if we substitute the implementation, they cannot rely on a specific subtype anyway.

Reply:
I removed that. It was for some convenience but in kotlin (as in modern java), it is not necessary.